Assignment 2 - Ryan McVay

ryan-mcvay73 opened the pull request.

ryan-mcvay73 committed:
Create workflows

ryan-mcvay73 committed:
Delete .github directory

ryan-mcvay73 committed:
Merge branch 'IntelliTect-Samples:main' into main

ryan-mcvay73 committed:
Create main.yml

ryan-mcvay73 committed:
Create Assignment1.yml

ryan-mcvay73 committed:
Delete .github/workflows/main.yml

ryan-mcvay73 committed:
Update Assignment1.yml

ryan-mcvay73 committed:
Update Assignment1.yml

ryan-mcvay73 committed:
Update Assignment1.yml

ryan-mcvay73 committed:
Update Assignment1.yml

ryan-mcvay73 committed:
Update readme.md

ryan-mcvay73 committed:
Add files via upload

ryan-mcvay73 committed:
Create Assignment2.ps1

ryan-mcvay73 committed:
Update readme.md

jenny-curry commented:
@ryan-mcvay73 Just checking if you added my identity (jcurry9@ewu.edu) to your assignment subscription with contributor role? I can’t see it in Azure.

ryan-mcvay73 commented:
My mistake, I had added ***@***.*** as a contributor and not your EWU email. I have now added the correct email address as a contributor. Thank you for letting me know.
________________________________
From: jenny-curry ***@***.***>
Sent: Monday, October 16, 2023 9:13 AM
To: IntelliTect-Samples/EWU-CSCD396-2023-Fall ***@***.***>
Cc: McVay, Ryan ***@***.***>; Mention ***@***.***>
Subject: Re: [IntelliTect-Samples/EWU-CSCD396-2023-Fall] Assignment 2 - Ryan McVay (PR #44)


@ryan-mcvay73<https://github.com/ryan-mcvay73> Just checking if you added my identity ***@***.******@***.***>) to your assignment subscription with contributor role? I can’t see it in Azure.

—
Reply to this email directly, view it on GitHub<https://github.com/IntelliTect-Samples/EWU-CSCD396-2023-Fall/pull/44#issuecomment-1764824292>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BC2DVZSAQA7EHWRHMNATWTLX7VMJPAVCNFSM6AAAAAA56KSUJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRUHAZDIMRZGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


CAUTION This email originated from outside your organization. Please exercise caution when clicking any links or opening attachments.

jenny-curry commented:
Secret should be referenced here. 
![image](https://github.com/IntelliTect-Samples/EWU-CSCD396-2023-Fall/assets/43355843/190d44a1-a130-4c6a-a452-9816bdabf5c7)

jenny-curry commented:
1. Create Web App
   {https://learn.microsoft.com/en-us/azure/app-service/quickstart-dotnetcore?tabs=net70&pivots=development-environment-cli}
   Note: Deploy application code using az cli, not the VSCode extension

- Web App Created ✅
  (You can use the below steps to publish your app OR use the 'az webapp up' command in the above tutorial)

  - Run command from your terminal 'dotnet publish SampleApp/MyFirstAzureWebApp'. This builds the application and files are genearted in SampleApp/MyFirstAzureWebApp/bin/Debug/net7.0/publish folder
  - Zip the items in your created publish folder
  - Use 'az webapp deploy' command to deploy your zip file to the application

- Url Accessible ✅

2. Create and Connect Storage Account - Using Azure CLI when option is available
   {https://learn.microsoft.com/en-us/azure/app-service/scenario-secure-app-access-storage?tabs=azure-cli}

- Enabled Managed Identity on Web App ✅
- Created Storage Account ✅
- Web App Granted Access to Storage Account ✅

3. Create a Key Vault
   https://learn.microsoft.com/en-us/azure/key-vault/general/quick-create-cli
   https://learn.microsoft.com/en-us/azure/app-service/app-service-configuration-references#granting-your-app-access-to-referenced-key-vaults
   https://learn.microsoft.com/en-us/azure/app-service/app-service-key-vault-references?tabs=azure-cli

- Key Vault Created ✅
- Key Vault Secret Created ✅
- Key Vault Access policy created for user 'jcurry9@ewu.edu' with secret 'Get' permission ✅
- Web App Granted Access to Key Vault ✅
- Reference key vault secret as an app setting on your web app where the app setting name is the secret name and the value references your secret using the secret uri ❌
  (https://learn.microsoft.com/en-us/azure/app-service/app-service-key-vault-references?tabs=azure-cli#source-app-settings-from-key-vault)

4. Create a PowerShell script called Assignment2.ps1 on your branch within the Assignment2 folder ✅
